src/server/scripts/EasternKingdoms/ZulGurub/boss_jeklik.cpp

```diff
@@ -265,45 +387,155 @@ struct boss_jeklik : public BossAI
     }
 };
 
-// Flying Bat
-struct npc_batrider : public ScriptedAI
+// Gurubashi Bat Rider (14750) - trash and boss summon are same creature ID
+struct npc_batrider : public CreatureAI
 {
-    npc_batrider(Creature* creature) : ScriptedAI(creature)
+    BatRiderMode _mode; // the version of this creature (trash or boss)
+
+    npc_batrider(Creature* creature) : CreatureAI(creature)
     {
-        _scheduler.SetValidator([this]
+        // if this is a summon of Jeklik, it is in boss mode
+        if
+        (
+            me->GetEntry() == NPC_BAT_RIDER &&
+            me->IsSummon() &&
+            me->ToTempSummon() &&
+            me->ToTempSummon()->GetSummoner() &&
+            me->ToTempSummon()->GetSummoner()->GetEntry() == NPC_PRIESTESS_JEKLIK
+        )
         {
-            return !me->HasUnitState(UNIT_STATE_CASTING);
-        });
+            LOG_DEBUG("scripts.ai", "npc_batrider::constructor: BAT_RIDER_MODE_BOSS");
+            _mode = BAT_RIDER_MODE_BOSS;
+
+            // make the bat rider unattackable
+            me->SetUnitFlag(UNIT_FLAG_NOT_SELECTABLE);
+            me->SetUnitFlag(UNIT_FLAG_IMMUNE_TO_PC);
+            me->SetUnitFlag(UNIT_FLAG_IMMUNE_TO_NPC);
+
+            // keep the bat from attacking players directly
+            me->SetReactState(REACT_PASSIVE);
+
+            // make the bat rider move the correct speed
+            me->SetSpeed(MOVE_WALK, 5.0f, true);
+
+        }
+        // otherwise, trash mode
+        else
+        {
+            LOG_DEBUG("scripts.ai", "npc_batrider::constructor: BAT_RIDER_MODE_TRASH");
+            me->SetReactState(REACT_DEFENSIVE);
+            _mode = BAT_RIDER_MODE_TRASH;
+        }
     }
 
     void Reset() override
     {
-        _scheduler.CancelAll();
-        me->SetUnitFlag(UNIT_FLAG_NOT_SELECTABLE);
-        me->SetHover(true);
-        me->SetDisableGravity(true);
-        me->AddUnitState(UNIT_STATE_ROOT);
+        CreatureAI::Reset();
+
+        switch (_mode)
+        {
+            case BAT_RIDER_MODE_BOSS:
+                events.Reset();
+                me->GetMotionMaster()->Clear();
+                break;
+            case BAT_RIDER_MODE_TRASH:
+                events.Reset();
+
+                // apply the Thrash (8876) spell to the bat rider (passive ability)
+                me->CastSpell(me, SPELL_PASSIVE_THRASH);
```

Comment:
add this as aura in the db. `creature_template_addon`

Reply:
Are you okay with the flying boss version (incorrectly) having this aura too then? Would it make a difference if I moved this to initialization and only cast it once?

Reply:
if the adds are pre-spawned, you can also potentially add it to `creature_addon` so it only applies to the creatures that are pre-spawned. but nah this is fine then imo